lib/src/boilerplate_suggestors/boilerplate_utilities.dart

```diff
@@ -413,3 +468,88 @@ extension IterableAstUtils on Iterable<NamedType> {
     }).join('${separator ?? ','} ');
   }
 }
+
+/// Adds [publicExportLocationsComment] to [classNode] or updates an existing comment.
+void addPublicExportLocationsComment(ClassDeclaration classNode,
+    SourceFile sourceFile, SemverHelper semverHelper, YieldPatch yieldPatch) {
+  final existingReportUnavailableCommentToken =
+      getComment(classNode, sourceFile, reportNotAvailableComment, yieldPatch);
+  final existingExportLocationsCommentToken =
+      getComment(classNode, sourceFile, classNotUpdatedComment, yieldPatch);
+  final exportLocationsComment =
+      publicExportLocationsComment(classNode, semverHelper);
+
+  if (hasComment(classNode, sourceFile, exportLocationsComment)) return;
+
+  if (existingReportUnavailableCommentToken != null) {
+    // Replace existing semver report unavailable with new comment.
+    yieldPatch(
+      existingReportUnavailableCommentToken.offset,
+      existingReportUnavailableCommentToken.end,
+      exportLocationsComment,
+    );
+  } else if (existingExportLocationsCommentToken != null) {
+    // Replace public export locations comment with new comment.
+    yieldPatch(
+      existingExportLocationsCommentToken.offset,
+      existingExportLocationsCommentToken.end,
+      exportLocationsComment,
+    );
+  } else {
+    // Add public export locations comment.
+    yieldPatch(
+      classNode.offset,
+      classNode.offset,
+      publicExportLocationsComment(classNode, semverHelper) + '\n',
+    );
+  }
+}
+
+/// Removes [publicExportLocationsComment] from [classNode].
+void removePublicExportLocationsComment(ClassDeclaration classNode,
+    SourceFile sourceFile, SemverHelper semverHelper, YieldPatch yieldPatch) {
+  final existingReportUnavailableCommentToken =
+      getComment(classNode, sourceFile, reportNotAvailableComment, yieldPatch);
+  final existingExportLocationsCommentToken =
+      getComment(classNode, sourceFile, classNotUpdatedComment, yieldPatch);
+
+  // Remove semver report unavailable comment.
+  if (existingReportUnavailableCommentToken != null) {
+    yieldPatch(
+      existingReportUnavailableCommentToken.offset,
+      existingReportUnavailableCommentToken.end,
+      '',
+    );
+  }
+
+  // Remove public export locations comment.
+  if (existingExportLocationsCommentToken != null) {
+    yieldPatch(
+      existingExportLocationsCommentToken.offset,
+      existingExportLocationsCommentToken.end,
+      '',
+    );
+  }
+}
+
+/// Returns AST token for [commentToGet] from the comments above [node].
+///
+/// If [commentToGet] does not exist above [node], returns null.
+Token getComment(AstNode node, SourceFile sourceFile, String commentToGet,
```

Comment:
I think this should live in the same file as hasComment (and hasComment could probably be DRY'd up to use the same code as `getComment`, since they're almost identical).